File: onebot-sdk/src/main/java/cn/evole/onebot/sdk/response/guild/GuildMemberProfileResp.java
package cn.evole.onebot.sdk.response.guild;

import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.List;

/**
 * @author cnlimiter
 */
@Data
public class GuildMemberProfileResp {

    @SerializedName("tiny_id")
    private String tinyId;

    @SerializedName("nickname")
    private String nickname;

    @SerializedName("avatar_url")
    private String avatarUrl;

    @SerializedName("join_time")
    private String joinTime;

    @SerializedName("roles")
    private List<RoleInfo> roles;

    @Data
    public static class RoleInfo {

        @SerializedName("role_id")
        private String roleId;

        @SerializedName("role_name")
        private String roleName;

    }

}

File: overflow-core/src/main/kotlin/cn/evolvefield/onebot/sdk/response/guild/GuildMemberProfileResp.kt
package cn.evolvefield.onebot.sdk.response.guild

import com.google.gson.annotations.SerializedName

class GuildMemberProfileResp {
    @SerializedName("tiny_id")
    var tinyId: String = ""

    @SerializedName("nickname")
    var nickname: String = ""

    @SerializedName("avatar_url")
    var avatarUrl: String = ""

    @SerializedName("join_time")
    var joinTime: String = ""

    @SerializedName("roles")
    var roles: List<RoleInfo> = mutableListOf()

    class RoleInfo {
        @SerializedName("role_id")
        var roleId = ""

        @SerializedName("role_name")
        var roleName = ""
    }
}
